CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -246,4 +246,43 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
     return Civi::$statics[__CLASS__]['entity_financial_account'][$financialTypeID][$relationTypeId];
   }
 
+  /**
+   * Get Financial Account type relations
+   *
+   * @param $flip bool
```

Comment:
The idea for flip is not basically array_flip the result array. Its basically to generate an error based on $links mapping. 
when $flip = FALSE
the $financialAccountLinks array will have account_relationship as key and financial_account_type_id as value
$flip = true
the $financialAccountLinks array will have array of account_relationship as value and financial_account_type_id as key